packages/legend-application-studio/src/stores/EditorGraphState.ts

```diff
@@ -1065,6 +1152,177 @@ export class EditorGraphState {
+          : this.editorStore.tabManagerState.currentTab;
+      const currentTabElementPath =
+        this.editorStore.tabManagerState.currentTab instanceof
+        ElementEditorState
+          ? this.editorStore.tabManagerState.currentTab.elementPath
+          : undefined;
+      /**
+       * We remove the current editor state so that we no longer let React displays the element that belongs to the old graph
+       * NOTE: this causes an UI flash, but this is in many way, acceptable since the user probably should know that we are
+       * refreshing the memory graph anyway.
+       *
+       * If this is really bothering, we can handle it by building mocked replica of the current editor state using stub element
+       * e.g. if the current editor is a class, we stub the class, create a new class editor state around it and copy over
+       * navigation information, etc.
+       */
+      this.editorStore.tabManagerState.closeAllTabs();
+      yield flowResult(graph_dispose(this.editorStore.graphManagerState.graph));
+
+      const graphBuildState = ActionState.create();
+      yield this.editorStore.graphManagerState.graphManager.buildLightGraph(
+        newGraph,
+        entities,
+        graphBuildState,
+        {
+          TEMPORARY__preserveSectionIndex:
+            this.editorStore.applicationStore.config.options
+              .TEMPORARY__preserveSectionIndex,
+          strict: this.enableStrictMode,
+        },
+      );
+
+      // Activity States
+      this.editorStore.globalTestRunnerState = new GlobalTestRunnerState(
+        this.editorStore,
+        this.editorStore.sdlcState,
+      );
+
+      // NOTE: build model generation entities every-time we rebuild the graph - should we do this?
+      const generationsBuildState = ActionState.create();
+      yield this.editorStore.graphManagerState.graphManager.buildGenerations(
+        newGraph,
+        this.graphGenerationState.generatedEntities,
+        generationsBuildState,
+      );
+
+      this.editorStore.graphManagerState.graph = newGraph;
+      this.editorStore.graphManagerState.graphBuildState = graphBuildState;
+      this.editorStore.graphManagerState.generationsBuildState =
+        generationsBuildState;
+
+      this.updateExplorerTree();
+
+      /**
+       * Re-build the editor states which were opened before from the information we have stored before
+       * creating the new graph
+       */
+      this.editorStore.tabManagerState.recoverTabs(
+        openedTabPaths,
+        openedGeneratedFileTabStates,
+        currentTabState,
+        currentTabElementPath,
+        false,
+      );
+
+      this.editorStore.applicationStore.log.info(
+        LogEvent.create(GRAPH_MANAGER_EVENT.GRAPH_UPDATED_AND_REBUILT),
+        '[TOTAL]',
+        Date.now() - startTime,
+        'ms',
+      );
+      this.isUpdatingGraph = false;
+    } catch (error) {
+      assertErrorThrown(error);
+      this.editorStore.applicationStore.log.error(
```

Comment:
if weare truly decoupling the two methods can we look through this logic to ensure it still makes sense. 
I see `updateGraphAndApplicationInTextMode` but in here we have a check `this.editorStore.isInFormMode`. 
Does the catch logic of this still make sense in the text mode context ?

Reply:
It wouldn't. I am removing this